model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -120,7 +120,6 @@ public RealmModel getRealm(String id) {
         for (String id : entities) {
             RealmModel realm = session.realms().getRealm(id);
             if (realm != null) realms.add(realm);
-            em.flush();
```

Comment:
Removing this flush inside the for loop decreased the whoamI request a lot....

Reply:
This is interesting point. See #6012, the call to `flush` reportedly improved the performance for bigger number of realms.  As I mentioned in the https://github.com/keycloak/keycloak/pull/6012#pullrequestreview-311328910 comment, I'd rather have no `flush` whatsoever in the JPA layer. Yet now I wonder which of the alternatives is better.
Could you please try the same patch without this `flush` and report the results here?

Reply:
Hi, @hmlnarik 
I've tested with 560 realms in KC 9.0.4 where the fix is originally committed:
* With flush: Takes around 11min
* Without flush: Takes around 1min
Please let me know if more results are needed. Thanks!